src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClient.cs

```diff
@@ -432,15 +432,20 @@ public PSRoleDefinition UpdateRoleDefinition(PSRoleDefinition roleDefinition)
 
             ValidateRoleDefinition(roleDefinition);
 
-			PSRoleDefinition fetchedRoleDefinition = null;
-			foreach (String scope in roleDefinition.AssignableScopes)
-			{
-				fetchedRoleDefinition = this.GetRoleDefinition(roleDefinitionId, scope);
-				if (fetchedRoleDefinition != null)
-				{
-					break;
-				}
-			}
+            PSRoleDefinition fetchedRoleDefinition = null;
+            foreach (String scope in roleDefinition.AssignableScopes)
+            {
+                try
```

Comment:
@darshanhs90 why is the try-catch being added?

Reply:
customers can add new scopes as part of the assignable scopes[irrespective of the position,either at the start or the end].
Assuming they add a new scope at the start,the getroledefinition call tries to get the roledefinition at the new scope and results in a Roledefinition doesnt exist. Hence when we add the try catch,we can check through all the assigned scopes if the Role defintiion exists or not.

Reply:
@darshanhs90 thanks for the explanation. This PR looks good once the change log is updated.